codegen/gql_code_builder/lib/data.dart

```diff
@@ -71,25 +77,49 @@ Library buildDataLibrary(
 }
 
 Map<String, SourceSelections> _fragmentMap(SourceNode source) => {
-      for (var def
+      for (final def
           in source.document.definitions.whereType<FragmentDefinitionNode>())
         def.name.value: SourceSelections(
           url: source.url,
           selections: def.selectionSet.selections,
         ),
-      for (var import in source.imports) ..._fragmentMap(import)
+      for (final import in source.imports) ..._fragmentMap(import)
     };
 
+Map<String, Set<String>> _possibleTypesMap(SourceNode source,
+    [Set<String>? visitedSource, Map<String, Set<String>>? possibleTypesMap]) {
+  visitedSource ??= {};
+  possibleTypesMap ??= {};
+
+  source.imports.forEach((s) {
+    if (!visitedSource!.contains(source.url)) {
+      visitedSource.add(source.url);
+      _possibleTypesMap(s, visitedSource, possibleTypesMap);
+    }
+  });
+
+  source.document.possibleTypesMap().forEach((key, value) {
+    possibleTypesMap![key] ??= {};
+    possibleTypesMap[key]!.addAll(value);
+  });
```

Comment:
I think this is already utilizing what you mentioned. Isn't it?
But additionally I made a recursion to make a global possibleTypes map thorough all imported graphql files.

Reply:
yes, sorry missed that.